fix #238 replace package object by all object

julien-truffaut opened the pull request:
you cannot deprecate a package object so this change is not binary compatible

aoiroaoino commented on core/src/main/scala/monocle/std/Tree.scala:
```diff
@@ -2,6 +2,8 @@ package monocle.std
 import monocle.function._
 import monocle.{Iso, Lens}
+import monocle.function.all._
+import monocle.std.stream._
```
maybe, `import monocle.std.stream._` is unused.

julien-truffaut replied:
I think it is used to bring in scope the `HeadOption` instance for `Stream` but you raise a good point, I think there is a compiler option to disallow unused import.

aoiroaoino replied:
I think you're correct. `headOption` instance is need bring in scope.
but, I get erros....  (add `-Ywarn-unused-import` option)

```scala
[aoino@aoino-local Monocle]$ sbt clean compile
...
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Either3.scala:5: Unused import
[error] import scalaz.{\/, -\/, \/-}
[error]                ^
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Tree.scala:6: Unused import
[error] import monocle.std.stream._
[error]                           ^
[error] two errors found
[error] (core/compile:compileIncremental) Compilation failed
[error] Total time: 35 s, completed 2015/09/08 0:18:44
```
(sorry, Either3's unused import error is my bad..., I will fix it soon...)

I have tried some pattern.

- delete `import monocle.std.stream._` only
```scala
[aoino@aoino-local Monocle]$ sbt clean compile
...
[success] Total time: 50 s, completed 2015/09/08 0:20:42
```

- delete `with StreamOptics` in `StdInstances` trait only
```scala
[aoino@aoino-local Monocle]$ sbt clean compile
...
[success] Total time: 43 s, completed 2015/09/08 0:22:08
```

- delete `import monocle.std.stream._` and `with StreamOptics`.
```scala
[aoino@aoino-local Monocle]$ sbt clean compile
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Tree.scala:32: Could not find an instance of Cons[Stream[scalaz.Tree[A]],scalaz.Tree[A]], please check Monocle instance location policy to find out which import is necessary
[error]       case xs    => Tree.node(tree.rootLabel, headOption[Stream[Tree[A]], Tree[A]].modify(_set(newLeaf))(xs) )
[error]                                                         ^
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Tree.scala:47: Could not find an instance of Snoc[Stream[scalaz.Tree[A]],scalaz.Tree[A]], please check Monocle instance location policy to find out which import is necessary
[error]       case xs    => Tree.node(tree.rootLabel, lastOption[Stream[Tree[A]], Tree[A]].modify(_set(newLeaf))(xs) )
[error]                                                         ^
[error] two errors found
[error] (core/compile:compileIncremental) Compilation failed
[error] Total time: 33 s, completed 2015/09/08 0:23:52
```


maybe, `Cons[Stream[Tree[A]], Tree[A]]` implicit value is resolved in `StdInstances` trait....?

julien-truffaut commented:
@aoiroaoino Thanks for your careful review. Turns out I had forgotten to delete the package in `mocle.std` that's why Stream` import in was not required.

I had a go with `-Ywarn-unused-import` but I got some nonsensical error, so I removed it. Maybe I did something wrong, we can always add it in a later PR if someone can make it work correctly :)

I added all traits to `MonocleSuite` hence we don't need so many imports in tests.

julien-truffaut commented:
@aoiroaoino do you think it is correct now?

aoiroaoino commented:
sorry, for the late reply...

I have checked it. And it's correct.
LGTM :+1:

aoiroaoino commented on core/src/main/scala/monocle/std/Either3.scala:
```diff
@@ -2,7 +2,7 @@ package monocle.std
 import monocle.{Prism, PPrism}
 import scalaz.{Either3, Left3, Middle3, Right3}
-import scalaz.{\/, -\/, \/-}
+import scalaz.{-\/, \/-}
```
thank you for fix it :)

julien-truffaut commented:
Thanks for the review!
